src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -391,11 +399,27 @@ data CTx = CTx
     , ctMeta          :: CTxMeta
     , ctInputAddrs    :: [CId Addr]
     , ctOutputAddrs   :: [CId Addr]
+    , ctIsOutgoing    :: Bool
     } deriving (Show, Generic, Typeable)
 
 txContainsTitle :: Text -> CTx -> Bool
 txContainsTitle search = isInfixOf (toLower search) . toLower . ctmTitle . ctMeta
 
+-- TODO [CSM-288] Rename?
+-- | In case of A -> A tranaction, we have to return two similar 'CTx's:
+-- one with 'ctIsOutgoing' set to /true/, one with the flag set to /false/.
+-- This type gathers these two together.
+data CTxs = CTxs
```

Comment:
I think this solution is ok. We could have also just use `[CTx]` as we did before. Now we have `[CTxs]` in searchHistory but we can easily fallback to `catMaybes $ concatMap (\tx -> [ctsOutgoing, ctsIncoming]) [CTxs]` .

This way [CTxs] -> [CTx] we would get the same representation in getHistory and searchHistory. So before returning [CTxs] in returnHistory I would convert it to [CTx]. (because we already have information is it outgoing or incoming in CTx datatype as a flag)

what do you think @Martoon-00 ?